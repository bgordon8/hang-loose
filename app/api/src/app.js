import express from 'express'

const app = express()

app.get("/", (req, res) => {
    res.status(200).json({
        message: "hello world!!!! thats whats up!!!"
    })
})

if (process.env.NODE_ENV !== "test") {
    app.listen(3000, () => {
        console.log('listening on port 3000...')
    })
}

export default app