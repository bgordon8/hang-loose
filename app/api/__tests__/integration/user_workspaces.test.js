import supertest from 'supertest'
import app from '../../src/app'

const req = supertest(app)

describe('routes : user_workspaces', () => {
  describe('routes : user_workspaces', () => {
    describe('GET /user_workspaces', () => {
      it('returns all user_workspaces', async () => {
        const res = await req.get('/user_workspaces')

        expect(res.status).toBe(200)
        expect(res.body).toHaveProperty('status')
        expect(res.body.status).toBe('success')
        expect(res.body).toHaveProperty('user_workspaces')
        expect(res.body.user_workspaces[0]).toHaveProperty('id')
        expect(res.body.user_workspaces[0].id).toBe(1)
        expect(res.body.user_workspaces[0]).toHaveProperty('workspaceId')
        expect(res.body.user_workspaces[0].workspaceId).toBe(1)
        expect(res.body.user_workspaces[0]).toHaveProperty('userId')
        expect(res.body.user_workspaces[0].userId).toBe(1)
        expect(res.body).toMatchSnapshot()
      })
    })
  })

  describe('GET /user_workspaces/:id', () => {
    it('returns user_workspace by id', async () => {
      const res = await req.get('/user_workspaces/1')

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('user_workspace')
      expect(res.body.user_workspace).toHaveProperty('id')
      expect(res.body.user_workspace.id).toBe(1)
      expect(res.body.user_workspace).toHaveProperty('workspaceId')
      expect(res.body.user_workspace.workspaceId).toBe(1)
      expect(res.body.user_workspace).toHaveProperty('userId')
      expect(res.body.user_workspace.userId).toBe(1)
      expect(res.body).toMatchSnapshot()
    })
  })

  //POST
  describe('POST /user_workspaces', () => {
    it('creates new user_workspace', async () => {
      const res = await req.post('/user_workspaces').send({
        workspaceId: 2,
        userId: 2,
      })
      console.log(res.body)
      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('user_workspace')
      expect(res.body.user_workspace).toHaveProperty('id')
      expect(res.body.user_workspace.id).toBe(4)
      expect(res.body.user_workspace).toHaveProperty('userId')
      expect(res.body.user_workspace.userId).toBe(2)
      expect(res.body.user_workspace).toHaveProperty('workspaceId')
      expect(res.body.user_workspace.workspaceId).toBe(2)
      expect(res.body).toMatchSnapshot({
        user_workspace: {
          created_at: expect.any(String),
          updated_at: expect.any(String),
        },
      })
    })
  })
})
