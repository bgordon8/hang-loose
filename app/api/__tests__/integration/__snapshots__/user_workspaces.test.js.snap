// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes : user_workspaces GET /user_workspaces/:id returns user_workspace by id 1`] = `
Object {
  "status": "success",
  "user_workspace": Object {
    "id": 1,
    "true": null,
    "userId": 1,
    "workspaceId": 1,
  },
}
`;

exports[`routes : user_workspaces routes : user_workspaces GET /user_workspaces returns all user_workspaces 1`] = `
Object {
  "status": "success",
  "user_workspaces": Array [
    Object {
      "id": 1,
      "true": null,
      "userId": 1,
      "workspaceId": 1,
    },
    Object {
      "id": 2,
      "true": null,
      "userId": 2,
      "workspaceId": 1,
    },
    Object {
      "id": 3,
      "true": null,
      "userId": 1,
      "workspaceId": 2,
    },
  ],
}
`;
